types/rpi-gpio/index.d.ts

```diff
@@ -0,0 +1,136 @@
+// Project: https://github.com/JamesBarwell/rpi-gpio.js#readme
+// Definitions by: Giles Roadnight <https://github.com/Roaders>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.0
+
+/// <reference types="node" />
+import { EventEmitter } from 'events';
+
+type MODE = 'mode_rpi' | 'mode_bcm';
+type DIR = 'in' | 'out' | 'low' | 'high';
+type EDGE = 'none' | 'rising' | 'falling' | 'both';
+type ValueCallback<T> = (err?: Error | null, value?: T) => void;
+type ErrorCallback = (err?: Error | null) => void;
+
+declare class Gpio extends EventEmitter {
+    constructor();
+
+    readonly DIR_IN: DIR;
+    readonly DIR_OUT: DIR;
+    readonly DIR_LOW: DIR;
+    readonly DIR_HIGH: DIR;
+    readonly MODE_RPI: MODE;
+    readonly MODE_BCM: MODE;
+    readonly EDGE_NONE: EDGE;
+    readonly EDGE_RISING: EDGE;
+    readonly EDGE_FALLING: EDGE;
+    readonly EDGE_BOTH: EDGE;
+    /**
+     * Set pin reference mode. Defaults to 'mode_rpi'.
+     *
+     * @param mode Pin reference mode, 'mode_rpi' or 'mode_bcm'
+     */
+    setMode(mode: MODE): void;
+    /**
+     * Setup a channel for use as an input or output
+     *
+     * @param  channel   Reference to the pin in the current mode's schema
+     * @param _direction The pin direction, either 'in' or 'out'
+     * @param _edge       edge Informs the GPIO chip if it needs to generate interrupts. Either 'none', 'rising', 'falling' or 'both'. Defaults to 'none'
+     * @param _onSetup  callback
+     */
+    setup(channel: number, onSetup?: ValueCallback<boolean>): void;
+    setup(channel: number, direction?: DIR, onSetup?: ValueCallback<boolean>): void;
+    setup(channel: number, direction?: DIR, edge?: EDGE, onSetup?: ValueCallback<boolean>): void;
+    /**
+     * Write a value to a channel
+     *
+     * @param channel The channel to write to
+     * @param value   If true, turns the channel on, else turns off
+     * @param cb      Optional callback
+     */
+    write(channel: number, value: boolean, cb?: ErrorCallback): void;
+    /**
+     * Write a value to a channel
+     *
+     * @param channel The channel to write to
+     * @param value   If true, turns the channel on, else turns off
+     * @param cb      Optional callback
+     */
+    output(channel: number, value: boolean, cb?: ErrorCallback): void;
+    /**
+     * Read a value from a channel
+     *
+     * @param channel The channel to read from
+     * @param cb      Callback which receives the channel's boolean value
+     */
+    read(channel: number, cb: ValueCallback<boolean>): void;
+    /**
+     * Read a value from a channel
+     *
+     * @param channel The channel to read from
+     * @param cb      Callback which receives the channel's boolean value
+     */
+    input(channel: number, cb: ValueCallback<boolean>): void;
+    /**
+     * Unexport any pins setup by this module
+     *
+     * @param cb Optional callback
+     */
+    destroy(cb: ErrorCallback): void;
```

Comment:
the api describes this as no args one:
https://github.com/JamesBarwell/rpi-gpio.js#destroy
but the source as optional:
https://github.com/JamesBarwell/rpi-gpio.js/blob/master/rpi-gpio.js#L310
```suggestion
    destroy(cb?: ErrorCallback): void;
```

Reply:
In this case the source is incorrect. If you do pass a callback you get errors. I will correct the docs comment